pyaptamer/aptanet/tests/test_aptanet.py

```diff
@@ -38,9 +38,41 @@ def test_pipeline_fit_and_predict(aptamer_seq, protein_seq):
 @pytest.mark.skipif(
     sys.version_info >= (3, 13), reason="skorch does not support Python 3.13"
 )
-@parametrize_with_checks([AptaNetClassifier()])
+@pytest.mark.parametrize("aptamer_seq, protein_seq", params)
+def test_pipeline_fit_and_predict_regression(aptamer_seq, protein_seq):
+    """
+    Test if Pipeline predictions are valid floats and shape matches input
+    for regression.
+    """
+    pipe = AptaNetPipeline(classifier=AptaNetPredictor(task="regression"))
+
+    X_raw = [(aptamer_seq, protein_seq) for _ in range(40)]
+    y = np.linspace(0, 1, 40).astype(np.float32)
+
+    pipe.fit(X_raw, y)
+    preds = pipe.predict(X_raw)
+
+    assert preds.shape == (40,)
+    assert np.issubdtype(preds.dtype, np.floating)
+
+
+@pytest.mark.skipif(
+    sys.version_info >= (3, 13), reason="skorch does not support Python 3.13"
+)
+@parametrize_with_checks([AptaNetPredictor()])
 def test_sklearn_compatible_estimator(estimator, check):
     """
     Run scikit-learn's compatibility checks on the AptaPipeline.
     """
     check(estimator)
+
+
+@pytest.mark.skipif(
+    sys.version_info >= (3, 13), reason="skorch does not support Python 3.13"
+)
+@parametrize_with_checks([AptaNetPredictor(task="regression")])
+def test_sklearn_compatible_estimator_regression(estimator, check):
```

Comment:
why are we not using one and the same test, and the list of aptanetpredictors with two elements?